GSoC-2025/movement:Calibrating confidence scores (Qianyi He).md

```diff
@@ -0,0 +1,83 @@
+- Implement a calibration scheme based on error distance thresholds to determine prediction success for confidence calculation.
+- Implement a calibration scheme based on error to compute new confidence scores (and experiment with multiple error-to-confidence mapping functions).
+- Test all added functionalities and write documentation for the new features.  
+- Provide an example use case document from a sample dataset.
+
+**Extended:**  
+- Design a benchmark function that uses basic datasets and existing confidence-based functionalities to evaluate the quality of confidence calibration methods based on the performance of downstream tasks.   
+- Conduct experiments based on the benchmark function to identify the optimal new confidence calibration scheme or highlight the strengths and weaknesses of each method. 
+
+### **Implementation Timeline**  
+(*~25-30 hours per week*)  
+
+| **Week**         | **Tasks**                                           |
+|------------------|----------------------------------------------------|
+| **Community Bonding** | Review existing confidence-related functions and datasets, design benchmark criteria, and explore error-to-confidence mapping methods. |
+| **Week 1-2**      | Implement and validate the log-log fitting method from keypoint-moseq. |
+| **Week 3**        | Extend to support ground-truth datasets with confidence scores. |
+| **Week 4-6**      | Implement a calibration method based on error distance thresholds and test all the implemented functionalities. |
+| **Week 7-8**      | Implement and evaluate multiple error-to-confidence mapping methods. |
+| **Week 9-10**     | Design and implement the benchmark function for evaluation. |
+| **Week 11**       | Conduct benchmark experiments and analyze calibration performance. |
+| **Week 12-13**    | Summarize results, finalize documentation, and prepare the final report. |
+
+### **Communication Plan**  
+I will communicate with my mentor via Zulip chat for daily questions and discussions, with weekly Zoom meetings for progress updates and addressing challenges.
+
+## Personal statement
+
+- **Past experienc.** 
+
+    I am currently a Master's student in Data Science at the University of Chicago. Previously, I worked at ByteDance as a data analyst, where I gained experience in machine learning and data processing. I have also studied data interaction techniques and developed several web pages to better present data. Additionally, I have conducted research on *real-time spatial positioning and attitude recognition based on Kalman filtering*, giving me a certain level of understanding of pose estimation frameworks. Although I have extensive programming experience, I am new to open source. The *movement* project is my first participation in the open-source community, where I submitted a pull request to implement a feature requested in an issue.
```

Comment:
Can you provide specific examples of your prior experience?

For example, what did you work on while at ByteDance that may be relevant to this project? Can you provide examples of data visualisation web pages you have designed and expand on the techniques you used? What kind of data interaction techniques are you familiar with?

Your research experience also feels relevant, can you provide a link or provide more details on what that work entailed? How did it relate to pose estimation frameworks?

How long have you been programming in Python? What other languages do you know, which ones are you most fluent in? How did you learn to contribute to open-source, which resources did you find useful?